Add Workspace name label next to room name in the chats list

bernhardoj opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
Add Workspace name label next to room name in the chats list

### Fixed Issues
<!---
1. Please replace GH_LINK with a URL link to the GitHub issue this Pull Request is fixing.
2. Please replace PROPOSAL: GH_LINK_ISSUE(COMMENT) with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>
$ https://github.com/Expensify/App/issues/<number-of-the-issue(comment)>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/11543
PROPOSAL: https://github.com/Expensify/App/issues/11543#issuecomment-1269566203


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
To test, you need to be at least in a workspace chat room.

1. Open chat list page
2. Verify every chat room named **#admin** or **#announce**
a. is not truncated
b. has a workspace name shown next to the chat room name
3. Verify long workspace name is truncated in one line

- [x] Verify that no errors appear in the JS console

### PR Review Checklist
<!--
This is a checklist for PR authors & reviewers. Please make sure to complete all tasks and check them off once you do, or else Expensify has the right not to merge your PR!
-->
#### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
5. Verify a modal appears displaying a preview of that image
--->

To test, you need to be at least in a workspace chat room.

1. Open chat list page
2. Verify every chat room named **#admin** or **#announce**
a. is not truncated
b. has a workspace name shown next to the chat room name
3. Verify long workspace name is truncated in one line

- [x] Verify that no errors appear in the JS console

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web
<!-- Insert screenshots of your changes on the web platform-->
| Compact  | Not Compact |
| ------------- | ------------- |
| ![web_compact](https://user-images.githubusercontent.com/50919443/196737379-bb41390a-0ce8-4641-b045-ca1cc761dab9.png)  | ![web_not_compact](https://user-images.githubusercontent.com/50919443/196737396-5ddbbea9-32e9-4580-9640-4b12ae6aee0e.png)  |

#### Mobile Web - Chrome
<!-- Insert screenshots of your changes on the web platform (from chrome mobile browser)-->
| Compact  | Not Compact |
| ------------- | ------------- |
| ![android_moweb_compact](https://user-images.githubusercontent.com/50919443/196739655-b30229de-37ca-4809-9306-61540076d786.jpg) | ![android_moweb_not_compact](https://user-images.githubusercontent.com/50919443/196739665-2c82730b-4bdd-4bd4-aae4-19b8b51e722d.jpg)  |

#### Mobile Web - Safari
<!-- Insert screenshots of your changes on the web platform (from safari mobile browser)-->
| Compact  | Not Compact |
| ------------- | ------------- |
| ![iphone_moweb_compact](https://user-images.githubusercontent.com/50919443/196745371-d2c2e10d-b62a-49a7-9937-af6f0d71ed8f.png)  | ![iphone_moweb_not_compact](https://user-images.githubusercontent.com/50919443/196745381-d4497752-e342-43c2-a64f-d54f9e268af4.png)  |

#### Desktop
<!-- Insert screenshots of your changes on the desktop platform-->
| Compact  | Not Compact |
| ------------- | ------------- |
| ![desktop_compact](https://user-images.githubusercontent.com/50919443/196737556-51380127-6aae-4926-b843-ee24557a475a.png)  | ![desktop_not_compact](https://user-images.githubusercontent.com/50919443/196737564-bdc9450b-2220-4f90-bc20-466e6e88eba8.png)  |

#### iOS
<!-- Insert screenshots of your changes on the iOS platform-->
| Compact  | Not Compact |
| ------------- | ------------- |
| ![iphone_compact](https://user-images.githubusercontent.com/50919443/196745482-4870ec2a-8fc5-40e7-91fd-77a047060547.png) | ![iphone_not_compact](https://user-images.githubusercontent.com/50919443/196745497-9539ef24-af21-470c-b62b-9f7c47078f79.png)  |

#### Android
<!-- Insert screenshots of your changes on the Android platform-->
| Compact  | Not Compact |
| ------------- | ------------- |
| ![android_compact](https://user-images.githubusercontent.com/50919443/196739771-23275f54-90d9-41cc-9f47-a0ecb359f655.jpg)  | ![android_not_compact](https://user-images.githubusercontent.com/50919443/196739774-e5703691-30c1-4c06-a01a-cc692fb4eeab.jpg) |

bernhardoj committed:
Add TextPill next to DisplayNames

bernhardoj committed:
simplify logic

bernhardoj commented:
**_I have read the CLA Document and I hereby sign the CLA_**

shawnborton commented:
This is looking really good! Just a few design details:
- The space between the room name and the pill should be just 4px
- The horizontal padding inside of the pill should be 8px (it might be already, but just confirming)
- The font size of the pill should be 11px, and we should already have a font size variable for this (micro? not sure what it's called)

bernhardoj committed:
Changes styling a bit

bernhardoj commented:
I have updated the code and the screenshots

shawnborton commented:
For this screenshot here:
<img width="247" alt="image" src="https://user-images.githubusercontent.com/2319350/195878375-105e00e9-f16c-4766-8558-579e51d492aa.png">


I don't think we want the `#announce` room to truncate unless it was truly wider than the total available horizontal width of that row. Let me know if that makes sense.

mananjadhav commented on src/components/LHNOptionsList/OptionRowLHN.js:
```diff
@@ -160,15 +164,22 @@ const OptionRowLHN = (props) => {
+                                        textStyles={displayNameStyle}
+                                        shouldUseFullTitle={optionItem.isChatRoom || optionItem.isPolicyExpenseChat}
+                                    />
+                                    {optionItem.isChatRoom && (
```
I am wondering if the check should be on `subtitle` here?

bernhardoj replied:
Do you mean we should check if the `subtitle` is empty or not?

mananjadhav replied:
yeah! I am not sure if the `subtitle` is set only for the workspace chat rooms. Can you confirm this please?

bernhardoj replied:
I checked and `subtitle` is also set for the workspace chat itself (chat type is **policy expense chat**) and the `subtitle` text is always **Workspace**. You can look at below image where I tried to change the condition to check whether `subtitle` is empty or not.

![327485](https://user-images.githubusercontent.com/50919443/196027273-173ff1f9-33c4-41be-b2d2-f513ef62fb58.jpg)

And here is the code to decide the subtitle text.
https://github.com/Expensify/App/blob/e38df100d62d197689bb16888a3ac7859f157681/src/libs/ReportUtils.js#L278-L299

If you look at the code, chat room will have subtitle if it's a 
1. default room
2. user created policy room
3. **policy expense chat**

And if we take a look at below code to check if it's a chat room or not
https://github.com/Expensify/App/blob/e38df100d62d197689bb16888a3ac7859f157681/src/libs/ReportUtils.js#L194-L202

we can see that it's a chat room if it's a
1. default room
2. user created policy room

So, if we want to achieve the same result as image above, then we need to check the `subtitle` contents. Otherwise, I think checking `isChatRoom` is enough.

bernhardoj replied:
So, what do you think? Should I check the subtitle or keep the current check with isChatRoom?

mananjadhav replied:
Lets stick to what you have right now `isChatRoom`. My bad to create unnecessary confusion.

bernhardoj replied:
Done

mananjadhav commented on src/styles/themes/default.js:
```diff
@@ -14,6 +14,7 @@ export default {
     border: colors.gray2,
     borderFocus: colors.blue,
     icon: colors.gray3,
+    iconLight: 'rgba(198, 201, 202, 0.5)',
```
Instead of doing this, we can stick to the existing colors.gray values and pass the style as `opacity: 0.5`

bernhardoj replied:
If we apply the `opacity` style to the text, it will affect the background color and the text itself. I was thinking that we could take the `icon` color and append the string with '80' (the hex will be #C6C9CA80), which is the hex code for the 50% opacity. So, it will look like this 
`StyleUtils.getBackgroundColorStyle(themeColors.icon + '80')`
but I think it looks ugly, that's why I decided to add a new color.

If we still want to apply the `opacity` style, I think we need to change the `TextPill` component a bit by wrapping the text with `View`.

What do you think?

mananjadhav replied:
I am fine with wrapping the `View` around `Text`.

bernhardoj replied:
Okay, so it turns out that wrapping it with `View` and apply the opacity to the `View` also affects the `Text`. So, I think the only way is to apply the opacity to the `backgroundColor`.

bernhardoj replied:
Should I keep the `iconLight` color?

shawnborton replied:
I agree with the concern about opacity. I think it's fine to add a new color value here, but let's make sure it's a color in our variables/theme files. 

I think it would be something like `pillBGHighlighted` or something like that. But then let's reference the icon color here but change the opacity. I'm not sure how this is done in RN, but in Sass we might write something like:

```
$pillBGHighlighted: transparentize($color-icon, 50%);
```

Let me know if that makes sense.

bernhardoj replied:
As far as I know, RN does not have such thing, but we can make our own implementation. I am thinking that maybe we could make a new function in `StyleUtils.js` let say `getBackgroundColorWithOpacityStyle` which takes the hex code as string and the opacity as number (0 to 1). Then, we will convert that hex to `rgb` numbers then return `rgba(r, g, b, opacity)`. It will look like this:
```
function getBackgroundColorWithOpacityStyle(background, opacity) {
    const r = ....
    const g = ....
    const b = .....
    return {
        backgroundColor: `rgba(${r}, ${g}, ${b}, ${opacity})`
    }
}
```

What do you think?

shawnborton replied:
I will lean on @mananjadhav and @marcochavezf to evaluate that, but I like the idea.

mananjadhav replied:
@bernhardoj Can we not just use `rgb(r, g, b, 0.5)` in the `colors.js` ?

bernhardoj replied:
Yes, we can remove the `iconLight` and use the new function above that I propose like this:
`StyleUtils.getBackgroundColorWithOpacityStyle(themeColors.icon, 0.5)`.

bernhardoj replied:
My newest commit currently uses above approach. Let me know if there is any concern.

mananjadhav commented on src/styles/styles.js:
```diff
@@ -2693,6 +2693,17 @@ const styles = {
     },
+
+    textPill: {
+        ellipsizeMode: 'end',
```
Can you attach screenshots with the ellipsizeMode? Say a longer workspace name?

bernhardoj replied:
Done. Updated the screenshots.

mananjadhav commented on src/components/TextPill.js:
```diff
@@ -0,0 +1,32 @@
+
+const propTypes = {
+    /** Text to show */
+    children: PropTypes.string,
```
Can we rename this to something related to text say `title`? `text`. etc.?

`children` is generally used for nodes or React components and looks misleading here.

bernhardoj replied:
Done.

mananjadhav commented on src/components/TextPill.js:
```diff
@@ -0,0 +1,32 @@
+};
+
+const defaultProps = {
+    children: '',
```
Instead of setting this empty, we should make the property mandatory

bernhardoj replied:
Done.

mananjadhav reviewed (changes requested):
Requested some smaller changes. @bernhardoj I would recommend add tests and screenshots for workspace name with ellipsis.

Also can we please add elaborate QA steps, this would involve, maybe identifying the `#announce` and `#admin` named chat?

bernhardoj committed:
Change props

bernhardoj committed:
Fix styling

bernhardoj commented:
> For this screenshot here: <img alt="image" width="247" src="https://user-images.githubusercontent.com/2319350/195878375-105e00e9-f16c-4766-8558-579e51d492aa.png">
> 
> I don't think we want the `#announce` room to truncate unless it was truly wider than the total available horizontal width of that row. Let me know if that makes sense.

Fixed. I also put some newest screenshots with long workspace name and updated the test steps.

mananjadhav commented on src/components/LHNOptionsList/OptionRowLHN.js:
```diff
@@ -160,15 +162,24 @@ const OptionRowLHN = (props) => {
+                                    {optionItem.isChatRoom && (
+                                        <TextPill
+                                            style={textPillStyle}
+                                            accessibilityLabel="Chat room name"
```
@marcochavezf @shawnborton Can one of you confirm if this is the correct `accessibilityLabel` ?

shawnborton replied:
Deferring to @marcochavezf as I am unfamiliar with this one.

marcochavezf replied:
Ah good question, since this only be used for Workspace name atm let's change the value to `Workspace name`

bernhardoj replied:
Got it.

bernhardoj replied:
Done

mananjadhav commented on src/styles/styles.js:
```diff
@@ -1185,14 +1185,11 @@ const styles = {
-        ...whiteSpace.noWrap,
-    },
-
-    optionDisplayNameCompact: {
```
Not sure I understand this change. Are we removing this style? This is used at two places (`OptionRowLHN`, and also `OptionRow`). I think removing this will break the styling in OptionRow.

bernhardoj replied:
Sorry, I missed that it is also being used at OptionRow component. I merge both styles to fix the #announce text got truncated on not compact mode. I will fix it later. Thanks for pointing it out!

bernhardoj replied:
Done

mananjadhav commented:
@bernhardoj Some more comments.

bernhardoj committed:
Fix styling

shawnborton commented:
Screenshots look pretty good to me! @bernhardoj could you also try pinning one of these rooms that has a super long name, in both compact and normal mode, and sharing screenshots please? Thanks!

bernhardoj commented:
@shawnborton I have updated the screenshots.

shawnborton commented:
Awesome, screenshots look good to me! Thanks @bernhardoj

mananjadhav reviewed (dismissed):
Thanks for the changes and being patient with this @bernhardoj.

🎀 👀 🎀 
C+ reviewed

mananjadhav commented:
Here's the checklist.


- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.


### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web
<img width="419" alt="web-worskspace-pill" src="https://user-images.githubusercontent.com/3069065/197074812-5f6435ae-fadd-44f9-a78c-ea0ee7edf121.png">

#### Mobile Web - Chrome
<img width="396" alt="mweb-android-worskspace-pill" src="https://user-images.githubusercontent.com/3069065/197074860-b082a981-a0e2-46c7-aa82-a9a05f205b98.png">

#### Mobile Web - Safari
<img width="366" alt="mweb-safari-worskspace-pill" src="https://user-images.githubusercontent.com/3069065/197074888-6cdeebc3-4336-4521-aef6-3e6e792a29e6.png">

#### Desktop
<img width="380" alt="desktop-workspace-pill" src="https://user-images.githubusercontent.com/3069065/197074902-f4804222-173b-459c-a949-5c1435a5ac7c.png">

#### iOS
<img width="391" alt="ios-workspace-pill" src="https://user-images.githubusercontent.com/3069065/197075528-c43fd768-0315-41c1-811a-7776e9352e8d.png">

#### Android
<img width="466" alt="android-workspace-pill" src="https://user-images.githubusercontent.com/3069065/197075014-bb8acaa1-c73c-4731-a46b-5d2d0d840e5b.png">

marcochavezf commented on src/components/LHNOptionsList/OptionRowLHN.js:
```diff
@@ -65,11 +66,12 @@ const OptionRowLHN = (props) => {
-    const displayNameStyle = StyleUtils.combineStyles(props.viewMode === CONST.OPTION_MODE.COMPACT
-        ? [styles.optionDisplayName, ...textUnreadStyle, styles.optionDisplayNameCompact, styles.mr2]
-        : [styles.optionDisplayName, ...textUnreadStyle], props.style);
+    const displayNameStyle = StyleUtils.combineStyles([styles.optionDisplayName, styles.optionDisplayNameCompact, ...textUnreadStyle], props.style);
```
Ah why are we merging `styles.optionDisplayName` and `styles.optionDisplayNameCompact` here? Originally we were only appending `styles.optionDisplayNameCompact`  if `props.viewMode === CONST.OPTION_MODE.COMPACT`

bernhardoj replied:
When the `viewMode` is **Compact**, we put the latest chat text next to `DisplayName` and we append the `style.optionDisplayNameCompact` to make sure DisplayName will take the whole available width. Now, we have `TextPill` next to `DisplayName`, that's why I append it no matter what the `viewMode` is (if we don't do this, the `DisplayName` will get truncated like @shawnborton previously mentioned).

bernhardoj replied:
Any update on this?

marcochavezf replied:
Oh thanks for the explanation @bernhardoj, sorry for the delay here

marcochavezf reviewed (commented):
Thank you guys looks mostly good to me! Just a comment related to `displayNameStyle`

bernhardoj committed:
Merge with main branch

bernhardoj committed:
Merge branch 'main' of https://github.com/Expensify/App into feat/#11543

marcochavezf reviewed (approved):
Thank you guys, LGTM

marcochavezf commented:
The PR checklists were already filled out, but the GH actions are failing because the text of one checkbox [changed](https://github.com/Expensify/App/blob/main/.github/PULL_REQUEST_TEMPLATE.md). So I'm going to merge it to unblock this change.

marcochavezf commented:
> @marcochavezf looks like this was merged without the checklist test passing. Please add a note explaining why this was done and remove the `Emergency` label if this is not an emergency.

https://github.com/Expensify/App/pull/11799#issuecomment-1292254357

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/3332937820) to staging by @marcochavezf in version: 1.2.20-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/3342265019) to production by @chiragsalian in version: 1.2.20-3 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
